### PR TITLE
WT-6710 Change default transaction isolation to snapshot

### PR DESCRIPTION
**Pull request by kommiharibabu**
The default session/transaction isoliation is changed from
read-committed to snapshot for all the user sessions. The
internal sessions still operate under read-committed isolation
mode.

### CODE REVIEW

**Review by raviprakashgiri29 — COMMENTED** [review]
I think this document https://source.wiredtiger.com/develop/cursors.html also needs to be changed because it mentions the `read-committed (the default)` isolation levels.

**Comment by kommiharibabu** [comment]
@raviprakashgiri29 I already fixed those changes in `src/docs/cursors.dox` and `src/docs/transactions.dox` files. The HTML files get generated from the above files. Please let me know if you find anything more that needs to be fixed apart from the existing changes.

**Comment by raviprakashgiri29** [comment]
> @raviprakashgiri29 I already fixed those changes in `src/docs/cursors.dox` and `src/docs/transactions.dox` files. The HTML files get generated from the above files. Please let me know if you find anything more that needs to be fixed apart from the existing changes.

Thanks @kommiharibabu . Sorry, I missed the documentation changes.

**Review by raviprakashgiri29 — APPROVED** [review]
Changes LGTM.

File: src/session/session_api.c [hunk 1]
```diff
@@ -2120,6 +2120,8 @@ __wt_open_internal_session(WT_CONNECTION_IMPL *conn, const char *name, bool open
      */
     F_SET(session, session_flags | WT_SESSION_INTERNAL);
 
+    /* Internal sessions default to read-committed isolation. */
+    session->isolation = session->txn->isolation = WT_ISO_READ_COMMITTED;
```
Comment (agorrod) [inline src/session/session_api.c:2124]: Ah - this is one of the reasons why this change is smaller than I'd expected. I think it's clever - did you try going down the other path and seeing which internal operations need read-committed versus just happen to be using it?

**Comment by agorrod** [comment]
@kommiharibabu could you run a full MongoDB patch build with these changes?
I'm surprised the fallout is so small - I thought more Python tests would fail when this change was made.

File: src/session/session_api.c (hunk 1 above)
Comment (kommiharibabu) [reply src/session/session_api.c]: I changed it for only one internal session - meta ckpt session and ran a WT patch build, found no issues.

**Comment by kommiharibabu** [comment]
@agorrod I ran a patch build with all internal sessions as read-committed mode and found no issues.
https://spruce.mongodb.com/version/5fa33040d1fe070539854a84/tasks?page=0

I ran the patch build again with the latest changes
https://spruce.mongodb.com/version/5fa4fa52c9ec44199cc23506/tasks

**Comment by agorrod** [comment]
Sorry I've been sitting on this - @sueloverso would you mind doing a review? I'm concerned about this change but don't know why - I should just be happy that it's simpler than i expected :)

File: src/include/txn_inline.h [hunk 2]
```diff
@@ -1236,7 +1236,8 @@ __wt_txn_update_check(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE
     txn = session->txn;
     txn_global = &S2C(session)->txn_global;
 
-    if (txn->isolation != WT_ISO_SNAPSHOT)
+    /* Don't check if transaction isolation is not snapshot or the table is metadata. */
+    if (txn->isolation != WT_ISO_SNAPSHOT || WT_IS_METADATA(cbt->dhandle))
```
Comment (sueloverso) [inline src/include/txn_inline.h:1240]: I'm not clear on why we need this change. We set the metadata transaction to read-committed below so it should succeed on the `txn->isolation != WT_ISO_SNAPSHOT` conditional already.

**Review by sueloverso — APPROVED** [review]
LGTM. I have one question.

**Review by sueloverso — CHANGES_REQUESTED** [review]
I just noticed that there are documentation changes needed here. In `src/docs` there are at least two places where we talk about read-committed being the default. @raviprakashgiri29 mentioned this and you indicated these are fixed, but neither current develop nor this branch fix the documentation. I do not see any `src/docs` changes in this PR.

**Review by sueloverso — APPROVED** [review]
Ignore any earlier comments about docs. I see those changes now.

File: src/include/txn_inline.h (hunk 2 above)
Comment (kommiharibabu) [reply src/include/txn_inline.h:1240]: The code flow doesn't come from the meta ckpt session, it is coming from bulk cursor close checkpoint. Added more details in Jira.
